sync_diff_inspector/diff.go

```diff
@@ -265,6 +281,112 @@ func (df *Diff) consume(ctx context.Context, rangeInfo *splitter.RangeInfo) (boo
 	return isEqual, nil
 }
 
+func (df *Diff) BinGenerate(ctx context.Context, targetSource source.Source, tableRange *splitter.RangeInfo, count int64) (*splitter.RangeInfo, error) {
+	if count <= splitThreshold {
+		return tableRange, nil
+	}
+	// TODO Find great index
+	tableDiff := targetSource.GetTable(tableRange.GetTableIndex())
+	indices := dbutil.FindAllIndex(tableDiff.Info)
+
+	var (
+		isEqual1, isEqual2 bool
+		count1, count2     int64
+	)
+	tableRange1 := tableRange.Copy()
+	tableRange2 := tableRange.Copy()
+	// if no index, do not split
+	if len(indices) == 0 {
+		return tableRange, nil
+	}
+	var index *model.IndexInfo
+	// using the index
+	for _, i := range indices {
+		if tableRange.IndexID == i.ID {
+			index = i
+			break
+		}
+	}
+	if index == nil {
```

Comment:
It's better to log an error, and return nil. so keep the integrity of origin logic